WebAppForSolocoProject/Controllers/HomeController.cs

```diff
@@ -23,46 +26,274 @@ public HomeController(IConfiguration configuration, OwnerData ownerData)
         }
 
         [HttpGet]
-        public IActionResult Create()
+        public async Task<IActionResult> Create()
         {
-            var model = new HomeCreateVM();
-            model.ownersList = ownerData.GetOwners();
-            model.basePath = ConfigurationManager.AppSettings["basePath"].ToString();
+            var model = new HomeCreateVM()
+            {
+                Logs = new List<string>()
+            };
+            await Task.Run(() =>
+            {
+                model.OwnersList = ownerData.GetOwners();
+                model.FolderList = model.OwnersList.First().QualityFolders;
+                model.BasePath = ConfigurationManager.AppSettings["basePath"].ToString();
+            });
             return View(model);
         }
 
         [HttpPost]
-        [ValidateAntiForgeryToken]
-        public IActionResult Create(HomeCreateVM model)
+        public async Task<IActionResult> Run(HomeCreateVM model)
+        {
+            await UpdateModel(model);
+            return View("Create", model);
+        }
+
+        public async Task<HomeCreateVM> CreateDirectories(HomeCreateVM model)
         {
-            model.ownersList = ownerData.GetOwners();
-            Owner owner = ownerData.GetOwners().FirstOrDefault(o => o.Name == model.SelectedOwner);
-            if (ModelState.IsValid)
+            if (model.BasePath == null)
             {
+                model.Logs.Add("You must provide a base path!");
+            }
+            else
+            { 
                 try
                 {
-                    foreach (var path in owner.Paths)
+                    await Task.Run(() =>
                     {
-                        string pathToCreate = model.basePath + "\\" + model.SelectedOwner + "\\" + path;
-                        if(Directory.Exists(pathToCreate))
+                        foreach (var key in model.SelectedOwner.FolderPathPairs.Keys)
                         {
-                            pathToCreate += " - directory already exist.";
+                            foreach (var path in model.SelectedOwner.FolderPathPairs.GetValueOrDefault(key))
```

Comment:
create separate method which will check if directory exists, if not then create otherwise just return log messege. SO this method could return string/string[] with log/logs.